appengine/deploy/build-and-deploy.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -eu -o pipefail
+
+PROJECT_ID=$1
+CONFIG_SECRETS=$2
+
+echo "Reading configs from cloud secret manager"
+gcloud --project=${PROJECT_ID} secrets versions access latest --secret="${CONFIG_SECRETS}" > vault.conf
+
+#  run the build
+echo "Running maven"
+mvn -DskipTests clean install package -f ../../pom.xml
+
+# bundling dependencies
+rm -fr lib
+mkdir -p lib
+mvn -f ../../pom.xml dependency:copy-dependencies -DoutputDirectory=./appengine/deploy/lib
+cp ../../target/DSMServer.jar .
+cp ../../src/main/resources/log4j.xml .
+
+echo "Downloading and configuring tcell"
+gsutil cat gs://ddp-tcell/tcell-1.11.0.tar.gz | tar -xf -
+gcloud --project=${PROJECT_ID} secrets versions access latest --secret="study-manager-tcell" >  tcell/tcell_agent.config
+
+# deploy to gae
+gcloud app deploy -q --stop-previous-version --promote --project ${PROJECT_ID} StudyManager.yaml
```

Comment:
One thing you might want to consider here is deleting older versions of deployed service. There is an overall limit in in the GCP project. On the Pepper side, right after we deploy, we try to remove any older versions. 
What service count look like right now in ddp-dev:
<img width="254" alt="Screen Shot 2020-07-23 at 7 11 41 AM" src="https://user-images.githubusercontent.com/29984380/88280521-f469ad00-ccb3-11ea-9a82-aae9af1dbcf3.png">

How we deal with it in CircleCi:
https://github.com/broadinstitute/ddp-study-server/blob/7c7843b080b0fd9289178c38ea9477025cc99a18/.circleci/config.yml#L294

Removing versions sometime simply fails. If that happens we just let it go and try to remove it again next time there is a deployment.